apps/ios/Shared/Views/Chat/ChatItemView.swift

```diff
@@ -130,60 +126,31 @@ struct ChatItemContentView<Content: View>: View {
         CIChatFeatureView(chatItem: chatItem, feature: feature, iconColor: iconColor)
     }
 
-    private func membersConnectedItemText() -> Text {
-        if case let .groupRcv(member) = chatItem.chatDir,
-           let prevItem = chatModel.getPrevChatItem(chatItem),
-           let prevMember = prevItem.memberConnected,
-           let membersConnectedText = membersConnectedText(connectedMemberNames(member, prevMember, prevItem)) {
-            return chatEventText(membersConnectedText, chatItem.timestampText)
+    private var membersConnectedItemText: Text {
+        if let t = membersConnectedText {
+            return chatEventText(t, chatItem.timestampText)
         } else {
             return eventItemViewText()
         }
-
-        func connectedMemberNames(_ member: GroupMember, _ prevMember: GroupMember, _ prevItem: ChatItem) -> [String] {
-            [member.chatViewName, prevMember.chatViewName] + collectPrevConnectedMemberNames(prevItem)
-        }
     }
 
-    private func collectPrevConnectedMemberNames(_ ci: ChatItem) -> [String] {
-        guard let prevItem = chatModel.getPrevChatItem(ci),
-              let memberConnected = prevItem.memberConnected else {
-            return []
-        }
-        let prevMemberNames = collectPrevConnectedMemberNames(prevItem)
-        return [memberConnected.chatViewName] + prevMemberNames
-    }
-
-    private func membersConnectedText(_ memberNames: [String]) -> String? {
-        if memberNames.count > 3 {
-            return String.localizedStringWithFormat(
-                NSLocalizedString("%@ and %d other members connected", comment: "<member_names> and <n >= 2> other members connected (plural)"),
-                Array(memberNames.prefix(2)).joined(separator: ", "),
-                memberNames.count - 2
-            )
-        } else if memberNames.count >= 2,
-                  let lastMemberName = memberNames.last {
-            return String.localizedStringWithFormat(
-                NSLocalizedString("%@ and %@ connected", comment: "<member_name(s)> and <member_name> connected (plural)"),
-                memberNames.dropLast().joined(separator: ", "),
-                lastMemberName
-            )
-        } else {
-            return nil
-        }
+    private var membersConnectedText: String? {
+        var ns = chatModel.getConnectedMemberNames(chatItem)
+        return ns.count > 3
+            ? "\(ns[0]), \(ns[1]) and \(ns.count - 2) other members connected"
+            : ns.count == 3
+            ? "\(ns[0] + ", " + ns[1]) and \(ns[2]) connected"
+            : ns.count == 2
+            ? "\(ns[0]) and \(ns[1]) connected"
```

Comment:
why these shouldn't be NSLocalizedString?

Reply:
the question is why the should be. These are simpler.